### PR TITLE
Update Samsung Internet Data, Updating folder javascript/

### PR DESCRIPTION
**Pull request by AdaRoseCannon**
This contains the JavaScript portions of #5028 it is dependent on #5034.

### CODE REVIEW

File: javascript/builtins/WeakMap.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
               "version_added": "8"
             },
             "samsunginternet_android": {
-              "version_added": true
+              "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/WeakMap.json]: WeakMap is probably supported 
Amazed by our enhanced linter catching this! :100: @vinyldarkscratch

File: javascript/builtins/WeakSet.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
               "version_added": "9"
             },
             "samsunginternet_android": {
-              "version_added": true
+              "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/WeakSet.json]: WeakSet is supported I would say

File: javascript/builtins/WeakMap.json (hunk 1 above)
Comment (AdaRoseCannon) [reply javascript/builtins/WeakMap.json]: Nice catch!!

File: javascript/builtins/WeakSet.json (hunk 2 above)
Comment (AdaRoseCannon) [reply javascript/builtins/WeakSet.json]: Yes, nice catch!

File: javascript/operators/optional_chaining.json [hunk 3]
```diff
@@ -78,4 +78,3 @@
     }
   }
 }
-
```
Comment (Elchi3) [inline javascript/operators/optional_chaining.json]: Can you undo this change? (add back new line at end of file)

Comment (AdaRoseCannon) [reply javascript/operators/optional_chaining.json]: How peculiar my script was explicitly re-adding it to files which did not have it, perhaps my editor stripped it when I made manual changes. 

I'll update it. Nice spot.

Comment (Elchi3) [reply javascript/operators/optional_chaining.json]: Thank you! I wonder if something is up this file. It is also a thing in another PR https://github.com/mdn/browser-compat-data/pull/5026#pullrequestreview-305531577 :eyes:

Comment (vinyldarkscratch) [reply javascript/operators/optional_chaining.json]: Oh, I found out what it is: there’s two spaces at the end of the file that are getting removed, not the newline.  I’ll be merging that other PR to fix the issues!

Comment (vinyldarkscratch) [reply javascript/operators/optional_chaining.json]: Merged the other PR now, so this is no longer an issue!

**Comment by AdaRoseCannon** [comment]
There is inconsistent data in the earliest versions regarding sub features and their parent features, I will unify on the parent version.
